### PR TITLE
TASK-7: Navigation Bar

### PR DESCRIPTION
**Pull request by cristian-aldea**
**Task**

#35

**Pre-requisites**

- [ ] React component hierarchy exists for devs to follow

**Acceptance Criteria**

- [ ] Nav bar shows all the buttons and each button is routed (but can lead to dead links)
- [ ] Documentation present for every new component and function
```
 * Description of Function/Component
 *
 * @param param1 Description of parameter 1
 * @param param2 Description of parameter 2
```
- [ ] Unit and UI tests are implemented for every component and function
- [ ] Must match [Mockup](https://www.figma.com/file/FJPXMYI4ULCnuCEZvPs67k/?node-id=534%3A6528)

**Notes**
- Removed e2e tests since they were failing and we don't currently have any user story implemented.

### CODE REVIEW

**Review by JimLau49 — DISMISSED** [review]
Nice work with the media query, sonar cloud needs to be fixed as well so that it can merge but other than that looking good !

Pre-requisites
✔ React component hierarchy exists for devs to follow

Acceptance Criteria

✔ Nav bar shows all the buttons and each button is routed (but can lead to dead links)
✔ Documentation present for every new component and function
✔ Unit and UI tests are implemented for every component and function
✔ Must match Mockup

**Review by JimLau49 — DISMISSED** [review]

File: apps/mull-ui/src/constants.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+/**
+ * Map of the current routes for the application
+ */
+export const ROUTES = {
+  HOME: '/home',
+  MAP: '/map',
+  NEW_EVENT: '/new-event',
```
Comment (RGPosadas) [inline apps/mull-ui/src/constants.ts]: Not blocking, but to match with how we named our EPICs/USs/requirements, I would change this name to `/create-event`

**Review by RGPosadas — CHANGES_REQUESTED** [review]
**Acceptance Criteria**

- [x] Nav bar shows all the buttons and each button is routed (but can lead to dead links)
- [x] Documentation present for every new component and function
- [x] Unit and UI tests are implemented for every component and function
- [ ] Must match [Mockup](https://www.figma.com/file/FJPXMYI4ULCnuCEZvPs67k/?node-id=534%3A6528)

**Blocking Comments**
1. Nice job so far! My only concern is that the icons are bigger compared to their mobile counterpart (desktop is perfectly fine though).
image

1. The icons are too close to the navbar border on mobile as well.
image

File: apps/mull-ui/src/constants.ts (hunk 1 above)
Comment (cristian-aldea) [reply apps/mull-ui/src/constants.ts]: Sure thing

**Review by r-saba — CHANGES_REQUESTED** [review]
Looking good, just need to support older browsers.

**Acceptance Criteria**

- [x] Nav bar shows all the buttons and each button is routed (but can lead to dead links)
- [x] Documentation present for every new component and function
- [x] Unit and UI tests are implemented for every component and function
- [ ] Must match [Mockup](https://www.figma.com/file/FJPXMYI4ULCnuCEZvPs67k/?node-id=534%3A6528)

How it looks on my 2019 browser
image

**Comment by cristian-aldea** [comment]
@RGPosadas @r-saba thanks for the nice catches. Feel free to let me know of other issues you find

**Review by RGPosadas — DISMISSED** [review]
LGTM! 

**Acceptance Criteria**

- [x] Nav bar shows all the buttons and each button is routed (but can lead to dead links)
- [x] Documentation present for every new component and function
- [x] Unit and UI tests are implemented for every component and function
- [x] Must match [Mockup](https://www.figma.com/file/FJPXMYI4ULCnuCEZvPs67k/?node-id=534%3A6528)

**Review by JimLau49 — DISMISSED** [review]

File: apps/mull-ui/src/constants.ts [hunk 2]
```diff
@@ -0,0 +1,10 @@
+/**
+ * Map of the current routes for the application
+ */
+export const ROUTES = {
```
Comment (ghiobi) [inline apps/mull-ui/src/constants.ts]: If they are constants and the values in the properties don't change, might as well use enums from TypeScript.

Comment (cristian-aldea) [reply apps/mull-ui/src/constants.ts]: Sure why not

**Comment by cristian-aldea** [comment]
@r-saba  @ghiobi I've addressed the issue you brought up, but please don't re-review just yet, I want to have a discussion about the icons in today's scrum.

**Review by r-saba — APPROVED** [review]
Fantastic work! LGTM! I am accepting even though they don't match the current mockups*.

Note* It does not match mockup, but matches what the mockups will look like once they are updated with the newly agreed upon navigation buttons.

**Acceptance Criteria**

- [x] Nav bar shows all the buttons and each button is routed (but can lead to dead links)
- [x] Documentation present for every new component and function
- [x] Unit and UI tests are implemented for every component and function
- [ ] Must match [Mockup](https://www.figma.com/file/FJPXMYI4ULCnuCEZvPs67k/?node-id=534%3A6528)

**Review by RGPosadas — APPROVED** [review]
Wow, buttons look so much sleeker!
My LTGM still stands.

**Review by JimLau49 — APPROVED** [review]
Previous comments still apply, good work 🚀

**Pull request merged**